src/services/blocks/BlocksService.ts

```diff
@@ -451,220 +474,118 @@ export class BlocksService extends AbstractService {
-	 */
-	private shouldCalcFee(specVersion: number, specName: string): boolean {
-		const minConfigVersion =
-			SidecarConfig.config.SUBSTRATE.MIN_CALC_FEE_VERSION || 0;
-
-		if (!minimumCalcFeeVersions[specName]) {
-			/**
-			 * If we are in a chain that is not polkadot, kusama, or westend
-			 */
-			return specVersion >= minConfigVersion;
-		} else {
-			return specVersion >= minimumCalcFeeVersions[specName];
-		}
-	}
+	// REVIEW NOTE
+	// Wether or not fee calculations is done should come from the config for the chain (I expanded
+	// the scope of the controller config to also include this). I don't think we should give end users
+	// this configuration option; a pr should be made on behalf on the chain maintainers to keep the
+	// controllerConfig (or future equivalent) up to date
+	// private shouldCalcFee(specVersion: number): boolean {
+	// }
 
 	/**
-	 * `getDecorateAndExtractWeight` is responsible for returning a object
-	 * with a key that represents the runtimeVersion of the fetched block with a value
-	 * that points to the weight data needed to calculate fees. This object will then
-	 * get set into our local cache.
+	 *	Get a formatted blockweight store value for the runtime corresponding to the given block hash.
 	 *
-	 * @param blockHash The parentParentHash used to retreive the metadata
-	 * @param api Used if the runtime matches the api and to also decorate the metadata
-	 * @param specVersion Used to set the runtime in the cache
-	 * @param specName Used to check the chain were connected to and see if its polkadot or kusama
+	 * @param api ApiPromise
+	 * @param blockHash Hash of a block in the runtime to get the extrinsic base weight(s) for
+	 * @returns formatted block weight store entry
 	 */
-	private async getDecorateAndExtractWeight(
+	private async getWeight(
 		api: ApiPromise,
-		blockHash: Hash,
-		specVersion: number,
-		specName: string
-	): Promise<MetaConstsCache> {
-		const extractedWeight: MetaConstsCache = {};
-
-		const decoratedKeyValues = {
-			decorated: {
-				consts: {
-					system: {},
+		blockHash: BlockHash
+	): Promise<WeightValue> {
+		const metadata = await api.rpc.state.getMetadata(blockHash);
+		const {
+			consts: { system },
+		} = expandMetadata(api.registry, metadata);
+
+		let weightValue;
+		if (((system.blockWeights as unknown) as BlockWeights)?.perClass) {
+			const {
+				normal,
+				operational,
+				mandatory,
+			} = ((system.blockWeights as unknown) as BlockWeights)?.perClass;
+
+			const perClass = {
+				normal: {
+					baseExtrinsic: normal.baseExtrinsic.toBigInt(),
 				},
-			},
-		};
+				operational: {
+					baseExtrinsic: operational.baseExtrinsic.toBigInt(),
+				},
+				mandatory: {
+					baseExtrinsic: mandatory.baseExtrinsic.toBigInt(),
+				},
+			};
 
-		// Set the runtime version key to `decoratedKeyValues`.
-		extractedWeight[specVersion] = decoratedKeyValues;
-
-		const doesCacheThisChain = specName === 'polkadot' || specName === 'kusama';
```

Comment:
We shouldn't need to be hard coding stuff like this since we have the configs for each chain